tools/replica/reorder_remd_traj.py

```diff
@@ -0,0 +1,571 @@
+        cur_pos = fobj.tell()
+        byteinds.append( [nframe+1, cur_pos] ) # dummy index for the EOF
+        
+        # write to file
+        np.savetxt(byteindfns[n], np.array(byteinds), fmt = "%d")
+        
+        # close the trajfile object
+        fobj.close()
+        
+        return
+
+
+def write_reordered_traj(temp_inds, byte_inds, outtemps, temps, 
+                         frametuple_dict, nprod, writefreq,
+                         outtrajfns, infobjs):
+    """
+    Reorders trajectories by temp. and writes them to disk
+    
+    :param temp_inds: list index of temps (in the list of all temps) for which 
+                      reordered trajs will be produced on this proc.
+    
+    :param byte_inds: dict containing the (previously stored) byte indices
+                      for each replica file (key = replica number)
+    
+    :param outtemps: list of all temps for which to produce reordered trajs.
+    
+    :param temps: list of all temps used in the REMD simulation.
+    
+    :param outtrajfns: list of filenames for output (ordered) trajs.
+    
+    :param frametuple_dict: dict containing a tuple (replica #, frame #) 
+                            for each temp.
+    
+    :param nprod: number of production timesteps. 
+                  Last (nprod / writefreq) frames 
+                  from the end will be written to disk.
+    
+    :param writefreq: traj dump frequency in LAMMPS
+    
+    :param infobjs: list of file pointers to input (unordered) trajs.
+    """
+    
+    nframes = int(nprod / writefreq)
+    
+    for n in temp_inds:
+        # open string-buffer and file
+        buf = IOBuffer()
+        of = readwrite(outtrajfns[n], mode = "wb")
+        
+        # get frames
+        abs_temp_ind = np.argmin( abs(temps - outtemps[n]) )
+        frametuple = frametuple_dict[abs_temp_ind][-nframes:]
+         
+        # write frames to buffer
+        if me == ROOT:
+            pb = tqdm(frametuple, 
+                  desc = ("Buffering trajectories for writing"),
+                  leave = True, position = ROOT + 2*me,
+                  unit = 'frame/replica', unit_scale = True)
+                  
+            iterable = pb
+        else:
+            iterable = frametuple
+        
+        for i, (rep, frame) in enumerate(iterable):
+            infobj = infobjs[rep]
+            start_ptr = int(byte_inds[rep][frame,1])
+            stop_ptr = int(byte_inds[rep][frame+1,1])
+            byte_len = stop_ptr - start_ptr
+            infobj.seek(start_ptr)
+            buf.write(infobj.read(byte_len))
+        if me == ROOT: pb.close()
+        
+        # write buffer to disk
+        if me == ROOT: print("Writing buffer to file")
+        of.write(buf.getvalue())
+        of.close()
+        buf.close()
+    
+    for i in infobjs: i.close()
+    
+    return
+        
+        
+def get_canonical_logw(enefn, frametuple_dict, temps, nprod, writefreq,
+                       kB = 0.001987):
```

Comment:
i feel that the default value for kB can be deleted since kB has this default value from argparse

Reply:
done.